src/gui/qgsexpressionbuilderwidget.cpp

```diff
@@ -440,6 +443,26 @@ void QgsExpressionBuilderWidget::loadRecent( const QString &collection )
   }
 }
 
+void QgsExpressionBuilderWidget::loadLayers()
+{
+  QMap<QString, QgsMapLayer *> layers = QgsProject::instance()->mapLayers();
```

Comment:
Well.. realistically it's unlikely that we'll even remove it from core for 3.0. So removing it from GUI is a 4.0+ goal. 

But anyway, we should stay on top of this (like you've done here) and try to make it easier for our future selves...